pkgs/top-level/python-packages.nix

```diff
@@ -2908,6 +2908,8 @@ in {
 
   ipython = if isPy27 then
     callPackage ../development/python-modules/ipython/5.nix { }
+  else if isPy36 then
+    callPackage ../development/python-modules/ipython/7.16.nix { }
```

Comment:
I'd be cautious & thoroughly test this if-elseif-else syntax in Nix. I've had troubles before where it doesn't evaluate as you would expect from other languages.

Reply:
I had the same concern, so I tested it by building `ipython` for Python 2.7, 3.6, and 3.7:

```
nix-repl> :b python27Packages.ipython 

this derivation produced the following outputs:
  out -> /nix/store/2ihnjhkzr2mj1in0l6wkm21xar4pvc2c-python2.7-ipython-5.8.0

nix-repl> :b python36Packages.ipython 

this derivation produced the following outputs:
  out -> /nix/store/a0i7j61c7vd33kwbkbg5jwqnla1qx0zm-python3.6-ipython-7.16.1

nix-repl> :b python37Packages.ipython 

this derivation produced the following outputs:
  out -> /nix/store/qdb4fjicn2xhp8f6d3brg33n45zfm7lx-python3.7-ipython-7.17.0
```

You can see from the pathnames that the correct `ipython` version numbers are getting selected for each Python version.